src/format/command.rs

```diff
@@ -114,10 +118,20 @@ pub fn format_stdin(_check: bool) -> Result<(), Error> {
 
     let formatted = crate::format::pretty(src.as_ref()).map_err(|error| Error::Parse {
         path: PathBuf::from("<standard input>"),
+        src: src.clone(),
         error,
-        src,
     })?;
 
-    print!("{}", formatted);
-    Ok(())
+    if check {
+        if src != formatted {
+            Err(Error::Format {
+                input: FormatInput::STDIN,
+            })
+        } else {
+            Ok(())
+        }
+    } else {
+        print!("{}", formatted);
+        Ok(())
+    }
```

Comment:
Can we move this `if` to the `run` function, to match the logic for formatting files?